### PR TITLE
PR: FlexLayout demo for Flex Basis

### LINKED ISSUE #73: FlexLayout create a demo that clearly deomstrates Flex Basis, and is very similar in design to the other demos on the page

### PR DESCRIPTION
**Pull request by bbrennanbasis**
This PR closes #73.
Created a demo to demonstrate the `setItemBasis()` method for `FlexLayout`.

### CODE REVIEW

File: src/main/code_snippets/flexlayout/Basis.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Button basisButton = new Button("Set basis", e -> {
```
Comment (hyyan) [inline src/main/code_snippets/flexlayout/Basis.txt:1]: @bbrennanbasis we are not using snippets anymore. It should be removed

**Comment by hyyan** [comment]
@bbrennanbasis Smart, I like it. One thing I’d suggest is that it’s not entirely clear what the user is supposed to interact with in the demo. I missed the button selection part. What’s the plan for this? Will you add instructions directly in the demo, or are you planning to update the documentation to explain how the demo works?

**Review by hyyan — CHANGES_REQUESTED** [review]

File: src/main/java/com/webforj/samples/views/flexlayout/FlexBasisView.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+import com.webforj.router.annotation.FrameTitle;
+import com.webforj.router.annotation.Route;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Route
+@FrameTitle("Flex Basis")
+public class FlexBasisView extends Composite<Div> {
```
Comment (hyyan) [inline src/main/java/com/webforj/samples/views/flexlayout/FlexBasisView.java:18]: Refactor this method to reduce its cognitive complexity. This means that not everything should go into the constructor. Move the event listeners to a separate method to help lower the complexity.

Comment (hyyan) [reply src/main/java/com/webforj/samples/views/flexlayout/FlexBasisView.java:18]: https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=94&id=webforj_webforj-docs-samples&open=AZK6TfL8zADAUu5a4nOK

File: src/main/code_snippets/flexlayout/Basis.txt (hunk 1 above)
Comment (MatthewHawkins) [reply src/main/code_snippets/flexlayout/Basis.txt:1]: @hyyan Just to verify, are we making snippets totally obsolete? If so, we should remove them, and the ability to expand/condense code from all demos.

Comment (hyyan) [reply src/main/code_snippets/flexlayout/Basis.txt:1]: Yes, Matthew, they are obsolete. I believe I recently opened an issue for this, and they should be removed. Maintaining them is too time-consuming and prone to errors.

**Comment by bbrennanbasis** [comment]
@hyyan I've added your revision requests. Additionally, I've added tool tip text to the main button that changes based on how many boxes are selected.

**Review by MatthewHawkins — CHANGES_REQUESTED** [review]
@bbrennanbasis Hey Ben - looks great! I just ran it myself and also missed that I was supposed to click on a box first. I would add something, either to the docs, or to the demo (the docs are probably the best place) mentioning what to do in the demo.

I don't have any overt changes for this program - go ahead and make a small addition to the FlexLayout article on the doc site saying something like "The above (or the following) demo allows you to select a box and change its basis...). Once you've done that, ask us for review again! 👍

**Review by hyyan — APPROVED** [review]

**Comment by hyyan** [comment]
@bbrennanbasis  An explanation of how the demo works still needs to be added to the Flex Layout documentation.

**Comment by bbrennanbasis** [comment]
@hyyan Added the explanation and the URL in webforj-docs [PR #295](https://github.com/webforj/webforj-docs/pull/295).

**Comment by hyyan** [comment]
@MatthewHawkins, please review this and check if it aligns with the plans for documenting the new configuration approach. It might be best to hold off until you've completed that.

**Review by hyyan — APPROVED** [review]

**Review by MatthewHawkins — APPROVED** [review]

**Pull request merged**